csvkit/utilities/csvsql.py

```diff
@@ -221,20 +221,20 @@ def _failsafe_main(self):
                     self.output_file.write('%s\n' % statement)
 
         if self.connection:
-            if self.args.query:
-                if os.path.exists(self.args.query):
-                    with open(self.args.query, 'r') as f:
-                        query = f.read()
-                else:
-                    query = self.args.query
+            if self.args.queries:
+                queries = []
+                for query in self.args.queries:
+                    if os.path.exists(query):
+                        with open(query, 'r') as f:
+                            query = f.read()
+                    queries += query.split(';')
 
                 # Execute the specified SQL queries.
-                queries = query.split(';')
                 rows = None
 
-                for q in queries:
-                    if q.strip():
-                        rows = self.connection.execute(q)
+                for query in queries:
+                    if query.strip():
+                        rows = self.connection.execute(query)
```

Comment:
This means that `rows` will be set to whatever the last query returns, right? Wouldn't it be better to store all the rows from all queries in a (flat) list?

Reply:
Yes, that is the existing functionality, so not a problem with the PR.